### PR TITLE
Fix name mapping mis-application and infinite loop

### PR DESCRIPTION
**Pull request by RobGeada**
Addesses https://issues.redhat.com/browse/RHOAIENG-1115 and https://github.com/trustyai-explainability/odh-trustyai-demos/issues/4

Name mappings were erroneously applying the _keys_ of the request as the new field names, not the _values_.

Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/trustyai-explainability/trustyai-explainability/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `FAI-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.3.x] FAI-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Review by tteofili — APPROVED** [review]

File: explainability-service/src/main/java/org/kie/trustyai/service/payloads/service/Schema.java [hunk 1]
```diff
@@ -50,9 +59,11 @@ public Map<String, String> getNameMapping() {
         return nameMapping;
     }
 
-    public void setNameMapping(Map<String, String> nameMapping) {
+    public void applyNewNameMapping(Map<String, String> nameMapping) {
         this.nameMapping = nameMapping;
+        LOG.info("applying name mapping");
         calculateNameMappedItems();
+        LOG.info(mappedItems.keySet());
```
Comment (danielezonca) [inline explainability-service/src/main/java/org/kie/trustyai/service/payloads/service/Schema.java]: Please remove this logging before merge (or make it `debug`)

File: explainability-service/src/main/java/org/kie/trustyai/service/payloads/service/Schema.java [hunk 2]
```diff
@@ -38,9 +42,14 @@ public Map<String, SchemaItem> getItems() {
     private void calculateNameMappedItems() {
         this.mappedItems = new HashMap<>(items);
         if (!nameMapping.isEmpty()) {
+            // for key, value pair in the name mapping
             for (Map.Entry<String, String> mapping : nameMapping.entrySet()) {
+
+                // if there is a corresponding key in the raw field names
```
Comment (danielezonca) [inline explainability-service/src/main/java/org/kie/trustyai/service/payloads/service/Schema.java:59]: Can you add a unit test?

File: explainability-service/src/main/java/org/kie/trustyai/service/payloads/service/Schema.java (hunk 1 above)
Comment (RobGeada) [reply explainability-service/src/main/java/org/kie/trustyai/service/payloads/service/Schema.java]: whoops

**Review by danielezonca — APPROVED** [review]
LGTM

**Pull request merged**